ts/packages/anchor-errors/src/index.ts

```diff
@@ -0,0 +1,275 @@
+// Instruction errors.
+
+/** 8 byte instruction identifier not provided. */
+export const ANCHOR_ERROR__INSTRUCTION_MISSING = 100 as const;
```

Comment:
I kept the `as const` because that's what we do on web3.js v2 so I assumed there was some obscure TS reason but let me investigate that a bit further.

Reply:
I _think_ this is necessary so that this assertion:

https://github.com/solana-labs/solana-web3.js/blob/71cba9915ce30b1bac11e4cac1afa0150e96536a/packages/errors/src/error.ts#L5-L17

…produces `SolanaError<{some actual code}>` and not just `SolanaError<number>`. It has to refine to the actual code, so that this works:

```ts
if (isSolanaError(e, SOLANA_ERROR__FOO)) {
  console.log(e.context.actually.related.to.the.foo.error);
}
```

Reply:
If y'all strip out the `const` and this still works then we're probably good.

https://github.com/solana-labs/solana-web3.js/blob/71cba9915ce30b1bac11e4cac1afa0150e96536a/packages/errors/src/__typetests__/error-typetest.ts#L39-L58